pkg/component/etcd/etcd_test.go

```diff
@@ -489,13 +501,56 @@ var _ = Describe("Etcd", func() {
 
 			return obj
 		}
+
+		expectedVPA = &vpaautoscalingv1.VerticalPodAutoscaler{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      vpaName,
+				Namespace: testNamespace,
+				Labels:    map[string]string{v1beta1constants.LabelRole: "etcd-vpa-main"},
+			},
+			Spec: vpaautoscalingv1.VerticalPodAutoscalerSpec{
+				TargetRef: &autoscalingv1.CrossVersionObjectReference{
+					Name: etcdName, Kind: "StatefulSet",
+					APIVersion: appsv1.SchemeGroupVersion.String(),
+				},
+				UpdatePolicy: &vpaautoscalingv1.PodUpdatePolicy{
+					UpdateMode: &vpaUpdateMode,
+				},
+				ResourcePolicy: &vpaautoscalingv1.PodResourcePolicy{
+					ContainerPolicies: []vpaautoscalingv1.ContainerResourcePolicy{
+						{
+							ContainerName: "etcd",
+							MinAllowed:    corev1.ResourceList{corev1.ResourceMemory: resource.MustParse("200M")},
+							MaxAllowed: corev1.ResourceList{
+								corev1.ResourceCPU:    resource.MustParse("4"),
+								corev1.ResourceMemory: resource.MustParse("30G"),
+							},
+							ControlledValues: &controlledValues,
+							Mode:             &containerPolicyAuto,
+						},
+						{
+							ContainerName:    "backup-restore",
+							Mode:             &containerPolicyOff,
+							ControlledValues: &controlledValues,
+						},
+					},
+				},
+			},
+		}
 	)
 
 	BeforeEach(func() {
-		ctrl = gomock.NewController(GinkgoT())
-		c = mockclient.NewMockClient(ctrl)
+		hvpaEnabled = true
+		caRotationPhase = ""
 		fakeClient = fakeclient.NewClientBuilder().WithScheme(kubernetesscheme.Scheme).Build()
 		sm = fakesecretsmanager.New(fakeClient, testNamespace)
+		backupConfig = nil
+		replicas = pointer.Int32(1)
+	})
+
+	JustBeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		c = mockclient.NewMockClient(ctrl)
```

Comment:
I introduced `BeforeEach` blocks down below (lines 1203ff) to customize some variables used in this `JustBeforeEach` block. With this pattern you can have the same setup logic and just have different configuration settings in each of the `Context` blocks.

ginkgo executes the `BeforeEach` blocks first, and only "just before" executing the actual test case it will run the `JustBeforeEach`.